packages/react-server/src/__tests__/ReactFlightAsyncDebugInfo-test.js

```diff
@@ -0,0 +1,355 @@
+      line,
+      col,
+      enclosingLine,
+      enclosingCol,
+    ]);
+  }
+  return copy;
+}
+
+function normalizeIOInfo(ioInfo) {
+  const {debugTask, debugStack, ...copy} = ioInfo;
+  if (ioInfo.stack) {
+    copy.stack = normalizeStack(ioInfo.stack);
+  }
+  if (typeof ioInfo.start === 'number') {
+    copy.start = 0;
+  }
+  if (typeof ioInfo.end === 'number') {
+    copy.end = 0;
+  }
+  return copy;
+}
+
+function normalizeDebugInfo(debugInfo) {
+  if (Array.isArray(debugInfo.stack)) {
+    const {debugTask, debugStack, ...copy} = debugInfo;
+    copy.stack = normalizeStack(debugInfo.stack);
+    if (debugInfo.owner) {
+      copy.owner = normalizeDebugInfo(debugInfo.owner);
+    }
+    if (debugInfo.awaited) {
+      copy.awaited = normalizeIOInfo(copy.awaited);
+    }
+    return copy;
+  } else if (typeof debugInfo.time === 'number') {
+    return {...debugInfo, time: 0};
+  } else if (debugInfo.awaited) {
+    return {...debugInfo, awaited: normalizeIOInfo(debugInfo.awaited)};
+  } else {
+    return debugInfo;
+  }
+}
+
+function getDebugInfo(obj) {
+  const debugInfo = obj._debugInfo;
+  if (debugInfo) {
+    const copy = [];
+    for (let i = 0; i < debugInfo.length; i++) {
+      copy.push(normalizeDebugInfo(debugInfo[i]));
+    }
+    return copy;
+  }
+  return debugInfo;
+}
+
+describe('ReactFlightAsyncDebugInfo', () => {
+  beforeEach(() => {
+    jest.resetModules();
+    jest.useRealTimers();
+    patchSetImmediate();
+    global.console = require('console');
+
+    jest.mock('react', () => require('react/react.react-server'));
+    jest.mock('react-server-dom-webpack/server', () =>
+      require('react-server-dom-webpack/server.node'),
+    );
+    ReactServerDOMServer = require('react-server-dom-webpack/server');
+
+    jest.resetModules();
+    jest.useRealTimers();
+    patchSetImmediate();
+
+    __unmockReact();
+    jest.unmock('react-server-dom-webpack/server');
+    jest.mock('react-server-dom-webpack/client', () =>
+      require('react-server-dom-webpack/client.node'),
+    );
+
+    React = require('react');
+    ReactServerDOMClient = require('react-server-dom-webpack/client');
+    Stream = require('stream');
+  });
+
+  function delay(timeout) {
+    return new Promise(resolve => {
+      setTimeout(resolve, timeout);
+    });
+  }
+
+  it('can track async information when awaited', async () => {
+    async function getData() {
+      await delay(1);
+      const promise = delay(2);
+      await Promise.all([promise]);
+      return 'hi';
+    }
+
+    async function Component() {
+      const result = await getData();
+      return result;
+    }
+
+    const stream = ReactServerDOMServer.renderToPipeableStream(<Component />);
+
+    const readable = new Stream.PassThrough(streamOptions);
+
+    const result = ReactServerDOMClient.createFromNodeStream(readable, {
+      moduleMap: {},
+      moduleLoading: {},
+    });
+    stream.pipe(readable);
+
+    expect(await result).toBe('hi');
+    /*
```

Comment:
I had to disable the tests because our testing infra is a mess.

Snapshots doesn't work with gating.
Inline snapshots can't be regenerated with hermes parser.
Line numbers are off in various environments.

Reply:
I managed to reenable the tests by deleting the copyright header, all comments and using gated inline snapshots and not the `@gate` feature.